src/main/java/org/javarosa/core/util/ArrayUtilities.java

```diff
@@ -33,6 +35,20 @@ public static boolean arraysEqual(byte[] array1, byte[] array2) {
     }
 
     /**
+     * Reverses an Array
+     */
+    public static <T> T[] reverse(Object[] myArray, Class<? extends T[]> tClass) {
```

Comment:
Usually I prefer adding a dependency to re-implementing functionality but definitely your call